js/ParticleModifiers.js

```diff
@@ -103,3 +107,26 @@ PHOTONS.EvenIntervalIndexModifier.prototype.update = function( particle, target
 
 }
 
+//=======================================
+// LoopingTimeIntervalIndex Modifier
+//=======================================
+
+PHOTONS.LoopingTimeIntervalIndexModifier = function( totalSteps, imagesPerSecond) {
+		PHOTONS.Modifier.call( this );
+		this.totalSteps = Math.floor( totalSteps || 1 );
+		this.timePerImage = 1 / imagesPerSecond;
+	}
+
+PHOTONS.LoopingTimeIntervalIndexModifier.prototype = Object.create( PHOTONS.Modifier.prototype );
+
+PHOTONS.LoopingTimeIntervalIndexModifier.prototype.update = function( particle, target ) {
+
+	// keep track of the atlas we started on so we can start randomly and progress normally afterwards
+	if ( particle.atlasStartIndex == undefined ) 
+		particle.atlasStartIndex = particle.atlasIndex.x;	

```

Comment:
I'm not sure attaching custom data to the Particle object is the best approach here. If more modifiers start using this approach down the road, the Particle object could get cluttered with arbitrary custom data, and there could even be name collisions. I think ideally you'd have a local hash table linking each particle to custom data you associated with it. I know this definitely a less efficient approach, but I feel like it's a cleaner design. To make this possible, each particle would need a unique ID (which they don't currently have). If you'd like to implement that, it would be great!